sources/packages/backend/apps/api/src/route-controllers/restriction/_tests_/e2e/restriction.students.controller.getLocationProgramInstitutionRestrictions.e2e-spec.ts

```diff
@@ -0,0 +1,145 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  FakeStudentUsersTypes,
+  getStudentToken,
+} from "../../../../testHelpers";
+import {
+  createE2EDataSources,
+  createFakeEducationProgram,
+  createFakeInstitution,
+  createFakeInstitutionLocation,
+  createFakeUser,
+  E2EDataSources,
+  RestrictionCode,
+  saveFakeInstitutionRestriction,
+} from "@sims/test-utils";
+import { InstitutionRestriction, RestrictionActionType } from "@sims/sims-db";
+
+describe("RestrictionStudentsController(e2e)-getLocationProgramInstitutionRestrictions", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+  });
+
+  it(
+    "Should get the active institution restrictions which are effective for the given program and institution location" +
+      " when there is one or more active institution restrictions for the provided program and institution location.",
+    async () => {
+      // Arrange
+      const institutionRestriction = await createInstitutionRestriction();
+      // Endpoint with location and program.
+      const endpoint = `/students/restriction/institution/location/${institutionRestriction.location.id}/program/${institutionRestriction.program.id}`;
```

Comment:
id use a base endpoint url to avoid duplication

Reply:
Yes, but as the locationid and program id are created at the test level I am keeping it as it is.